src/cmd.ts

```diff
@@ -44,22 +61,117 @@ async function uploadAndRunAll(loginData: config.LoginData, folder: string, pref
+                uploadPath = uploadPath.replace("/**", "");
+            }
+
+            // if the path contains a wildcard, we have to make sure that wildcards can only be used at the end of a path
+            indexOf = uploadPath.lastIndexOf("/");
+            let wildcardIndex = uploadPath.indexOf("*");
+            let wildcard = "";
+            let wildcardReg: RegExp;
+
+            if (wildcardIndex > -1 && wildcardIndex < indexOf) {
+                throw new Error("Wildcards can only be used at the end of the path: " + wildcard);
+            } else {
+                wildcard = uploadPath.substr(indexOf + 1).split(".").join("\\.").split("*").join(".*");
+                uploadPath = uploadPath.substr(0, indexOf);
+
+                try {
+                    wildcardReg = new RegExp("^" + wildcard + "$");
+                } catch (err) {
+                    throw new Error("Invalid wildcard expression: " + wildcard);
+                }
+            }
+
+            let files = readDirSync(uploadPath, true);
+            let filesToUpload: sdsAccess.scriptT[] = [];
+
+            files.forEach(function (file) {
+                let parsedFile = path.parse(file);
+
+                if (wildcardReg.test(parsedFile.base)) {
+                    filesToUpload.push({
+                        name: parsedFile.name,
+                        path: file,
+                        sourceCode: fs.readFileSync(file).toString()
+                    });
+                }
+            });
+
+            if (filesToUpload.length < 0) {
+                console.log("No files to upload found.");
+            } else {
+                let loginData: config.LoginData = new config.LoginData();
+                loginData.loadConfigFile(json);
+                uploadAndRunAll(loginData, filesToUpload);
+            }
         }
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+        // if (dir.length > 0) {
+        //     let loginData: config.LoginData = new config.LoginData();
+        //     loginData.loadConfigFile(json)
+        //     let dirToUpload = dir[0];
+        //     let execPrefix = '';
+
+        //     // dirToUpload has to be a directory
+        //     if (fs.statSync(dirToUpload).isFile()) {
+        //         if (!path.isAbsolute(dirToUpload)) {
+        //             dirToUpload = path.join(process.cwd(), dirToUpload);
+        //         }
+
+        //         let parsedFile = path.parse(dirToUpload);
+        //         dirToUpload = parsedFile.dir;
+        //         execPrefix = parsedFile.base;
+        //     } else if (dir.length > 1) {
+        //         execPrefix = dir[1];
+        //     }
+
+        //     // upload and execute
+        //     console.log(`upload and execute scripts from dir '${dirToUpload}' with file prefix '${execPrefix}'`);
+        //     uploadAndRunAll(loginData, dirToUpload, execPrefix);
+        // } else {
+        //     console.log('Dir/file to run/upload missing');
+        // }
     });
```

Comment:
You should not commit commented-out lines. Just remove them. Use git for version control.